library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java

```diff
@@ -998,6 +1073,33 @@ public void setMaxRetryCount(int maxRetryCount)
   }
 
   /**
+   * This class scans a directory and put the list of files in a blocking queue.
+   */
+  public class DirectoryScannerThread implements Runnable
```

Comment:
Suggest renaming this DirectoryScannerRunnable, DirectoryScannerTask, or just DirectoryScanner since it does not create a new thread for each instance -- the same thread in the Executor runs all the instances.